submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoMetaQueryServiceTest.kt

```diff
@@ -0,0 +1,123 @@
+import java.time.Instant
+
+@ExtendWith(SpringExtension::class)
+@Testcontainers
+@SpringBootTest(classes = [MongoDbReposConfig::class])
+internal class SubmissionMongoMetaQueryServiceTest {
+
+    @Autowired
+    lateinit var testInstance: SubmissionMongoMetaQueryService
+
+    @Autowired
+    lateinit var submissionMongoRepository: SubmissionMongoRepository
+
+    @Test
+    fun getBasicProject() {
+        submissionMongoRepository.save(testDocSubmission(
+            accNo = "accNo1",
+            version = 1,
+            status = DocProcessingStatus.PROCESSED,
+            attributes = listOf(DocAttribute(SubFields.ACC_NO_TEMPLATE.value, "template"))
+        ))
+
+        val result = testInstance.getBasicProject("accNo1")
+
+        assertThat(result.accNo).isEqualTo("accNo1")
+    }
+
+    @Test
+    fun findLatestBasicByAccNo() {
+        submissionMongoRepository.save(testDocSubmission("accNo2", 1, DocProcessingStatus.PROCESSED))
+        submissionMongoRepository.save(testDocSubmission("accNo2", -2, DocProcessingStatus.PROCESSED))
+        submissionMongoRepository.save(testDocSubmission("accNo2", 4, DocProcessingStatus.PROCESSED))
+
+        val lastVersion = testInstance.findLatestBasicByAccNo("accNo2")
+
+        assertThat(lastVersion).isNotNull()
+        assertThat(lastVersion!!.version).isEqualTo(4)
+    }
+
+    @Test
+    fun getAccessTags() {
+    }
+
+    @Test
+    fun `exists by AccNo when exists`() {
+        submissionMongoRepository.save(testDocSubmission("accNo3", 1, DocProcessingStatus.PROCESSED))
+
+        assertThat(submissionMongoRepository.existsByAccNo("accNo3")).isTrue()
+    }
+
+    @Test
+    fun `exist by AccNo when don't exists`() {
+        submissionMongoRepository.save(testDocSubmission("accNo4", 1, DocProcessingStatus.PROCESSED))
+
+        assertThat(submissionMongoRepository.existsByAccNo("accNo5")).isFalse()
+    }
+
+    private fun testDocSubmission(
+        accNo: String,
+        version: Int,
+        status: DocProcessingStatus,
+        attributes: List<DocAttribute> = listOf()
+    ): DocSubmission {
+        return DocSubmission(
+            id = "",
+            accNo = accNo,
+            version = version,
+            owner = "",
+            submitter = "",
+            title = "",
+            method = DocSubmissionMethod.PAGE_TAB,
+            relPath = "",
+            rootPath = "",
+            released = true,
+            secretKey = "",
+            status = status,
+            releaseTime = Instant.ofEpochSecond(1),
+            modificationTime = Instant.ofEpochSecond(2),
+            creationTime = Instant.ofEpochSecond(3),
+            section = DocSection(type = ""),
+            attributes = attributes,
+            tags = listOf(),
+            projects = listOf(),
+            stats = listOf()
+        )
+    }
+
+    companion object {
+
+        @Container
+        val mongoContainer: MongoDBContainer = MongoDBContainer(DockerImageName.parse("mongo:4.0.10"))
```

Comment:
- Is this the mongo version we should be using and is it the same as the version provided by IT?
- Maybe we should have this in some sort of constant. Not sure if the versions declared [here](https://github.com/EBIBioStudies/biostudies-backend-services/blob/master/buildSrc/src/main/kotlin/Dependencies.kt#L59) can be referenced in the code 🤔